dash/pages/cohorts.py

```diff
@@ -0,0 +1,199 @@
+                // TODO - check range sliders
+                if (termCountLookup[`${p}-${h.key}`]) {
+                    termCountLookup[`${p}-${h.key}`].innerText = h.count
+                    // remove from array
+                    delete termCountLookup[`${p}-${h.key}`]
+                }
+              }) ;
+              // set items no longer in histogram to 0
+              Object.keys(termCountLookup).forEach((k) => {
+                if (k.startsWith(p)) {
+                    termCountLookup[k].innerText = '0';
+                }          
+              }) ;                    
+            }                 
+        });
+        // always return no update since we updated dom directly
+        return window.dash_clientside.no_update
+    }
+    """,
+    Output('placeholder-dummy', 'children'),
+    Input('histogram-data', 'data'),
+    prevent_initial_call=True
+)
+
+
+def layout():
+    """Render the cohort page."""
+
+    def accordian(items_, dfs_, names_):
+        """Create an accordian with items for each facet."""
+        return html.Div(dbc.Accordion(
+            [
+                dbc.AccordionItem(
+                    [
+                        item,
+                    ],
+                    title=f"{titleize(pluralize(name))}"
+                )
+                for item, df, name in zip(items_, dfs_, names_)
+            ],
+            start_collapsed=True,
+            always_open=True
+        ))
+
+    # get the data from guppy and plot each aggregation
+    file_histograms = get_file_histograms()
+
+    items = []
+    data_frames = []
+    names = []
+
+    for i, d, n in [histogram_selects(file_histograms), histogram_sliders(file_histograms)]:
+        items.extend(i)
+        data_frames.extend(d)
+        names.extend(n)
+
+    file_accordian = accordian(items, data_frames, names)
+
+    observation_histograms = get_observation_histograms()
+
+    items = []
+    data_frames = []
+    names = []
+
+    for i, d, n in [histogram_selects(observation_histograms), histogram_sliders(observation_histograms)]:
+        items.extend(i)
+        data_frames.extend(d)
+        names.extend(n)
+
+    observation_accordian = accordian(items, data_frames, names)
+
+    return [
+        html.H2("Cohorts"),
+        html.Hr(className="my-2"),
+        html.P('Quis imperdiet massa tincidunt nunc. Convallis tellus id interdum velit. Mauris pellentesque pulvinar pellentesque habitant morbi tristique senectus.'),
+        html.Hr(className="my-2"),
+        html.Code(
+            id="query-builder",
+            children="Selections go here..."),
+        html.Hr(className="my-2"),
+        html.Button('Query', id='query', n_clicks=0, style={'display': 'flex', 'float': 'right'}),
+        dbc.Tabs(
+            [
+                dbc.Tab(file_accordian, label="Files"),
+                dbc.Tab(observation_accordian, label="Observations"),
+            ]
+        ),
+        dash_table.DataTable(id='results'),
```

Comment:
Interesting how some data figures can be called from Dash like DataTable while others like the Histogram have to be assembled more by hand